app/routes.py

```diff
@@ -1,2 +1,184 @@
+
+######################## VIDEO ENDPOINTS ######################## 
+
+@videos_bp.route("/", methods = ["POST"], strict_slashes=False)
+def register_video():
+    request_body = request.get_json()
+    if invalid_video_request(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    new_video = Video(title=request_body["title"],
+                        release_date=request_body["release_date"],
+                        total_inventory=request_body["total_inventory"],
+                        available_inventory=request_body["total_inventory"])
+    db.session.add(new_video)
+    db.session.commit()
+    return jsonify(id=new_video.id), 201
+
+@videos_bp.route("/", methods = ["GET"], strict_slashes=False)
+def view_all_videos():
+    videos = Video.query.all()
+    all_videos = [video.to_json() for video in videos]
+    return jsonify(all_videos)
+
+@videos_bp.route("/<int:video_id>", methods = ["GET"], strict_slashes=False)
+def view_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    return jsonify(video.to_json())
+
+@videos_bp.route("/<int:video_id>", methods = ["DELETE"], strict_slashes=False)
+def delete_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    db.session.delete(video)
+    db.session.commit()
+    return jsonify(id=video.id)
+
+@videos_bp.route("/<int:video_id>", methods = ["PUT"], strict_slashes=False)
+def update_video(video_id):
+    video = Video.query.get_or_404(video_id)
+    request_body = request.get_json()
+    if invalid_video_request(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    video.title=request_body["title"]
+    video.release_date=request_body["release_date"]
+    video.total_inventory=request_body["total_inventory"]
+    db.session.commit()
+    return jsonify(video.to_json())
+
+@videos_bp.route("/<int:video_id>/rentals", methods = ["GET"], strict_slashes=False)
+def view_customer_rentals(video_id):
+    video = Video.query.get_or_404(video_id)
+    video_rentals = []
+    for rental in video.customers:
+        customer = Customer.query.get(rental.customer_id)
+        customer_details = {"due_date": rental.due_date,
+                            "name": customer.name,
+                            "phone": customer.phone,
+                            "postal_code": customer.postal_code}
+        video_rentals.append(customer_details)
+    return jsonify(video_rentals)
+
+######################## RENTAL ENDPOINTS ######################## 
+
+@rentals_bp.route("/check-out", methods = ["POST"], strict_slashes=False)
+def checkout():
+    request_body = request.get_json()
+    if invalid_checkout_checkin(request_body):
+        return jsonify(details=f"Invalid data"), 400
+    new_rental = Rental(customer_id = request_body["customer_id"],
+                        video_id = request_body["video_id"])
+    customer = Customer.query.get(request_body["customer_id"])
+    customer.videos_checked_out_count += 1
```

Comment:
This line should go after our check for available inventory. If we do not have any available inventory, our `videos_checked_out_count` should not increase. This is an example of how we could end of with conflicting data in our database,